frontend/components/product-list/sections/FilterableProductsSection/RangeInput.tsx

```diff
@@ -0,0 +1,215 @@
+import {
+   Box,
+   FormControl,
+   FormErrorMessage,
+   FormLabel,
+   HStack,
+   Input,
+   InputGroup,
+   InputLeftElement,
+   Text,
+} from '@chakra-ui/react';
+import { formatFacetName } from '@helpers/algolia-helpers';
+import { useDebounce } from '@ifixit/ui';
+import * as React from 'react';
+import { useRange, UseRangeProps } from 'react-instantsearch-hooks-web';
+import { useSearchCache } from './useSearchCache';
+
+export type RangeInputProps = UseRangeProps;
+
+const DEBOUNCE_INTERVAL_MILLIS = 300;
+
+export function RangeInput(props: RangeInputProps) {
+   const { refresh } = useSearchCache();
+   // This is a quick hack to get `useRange` to work. Without this, `useRange`
+   // would behave like price is not refinable. Eventually we should figure out
+   // how to make `useRange` work without this hack.
+   React.useEffect(() => {
+      refresh();
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+   }, []);
+
+   const range = useRange(props);
+   const appliedRangeMin = range.start[0];
+   const appliedRangeMax = range.start[1];
+
+   const refineMinValue = React.useCallback(
+      (value: number | undefined) => {
+         const minAllowedValue = range.range.min ? range.range.min + 1 : 0;
```

Comment:
I don't get the `min - 1` here...  The result is that if the cheapest product is $4 and you type in `4` in the `min` part of the range it'll be immediately changed to `5`. This feels like a bug.

Reply:
The behaviour of the Algolia `useRange` is tricky and it's optimized to limit refinement within the existing price range, in order to avoid showing empty results. 
In this specific case, if you input a number at the boundaries (or outside the range) it won't do anything, because when the lowest price option is `4` and you put `4` is the same thing as not adding the filter at all.

I think you're right though, my current implementation can feel buggy.  I've invested a little bit more time, and I think I've found a better trade off. I'll push the changes shortly, let me know what you think.